src/main/java/org/folio/service/manager/export/strategy/InstanceExportStrategyImpl.java

```diff
@@ -52,25 +55,34 @@ public void export(ExportPayload exportPayload, Promise<Object> blockingPromise)
     MappingProfile mappingProfile = exportPayload.getMappingProfile();
     OkapiConnectionParams params = exportPayload.getOkapiConnectionParams();
 
-    if (Objects.nonNull(mappingProfile)
-      && mappingProfile.getRecordTypes().contains(RecordType.SRS)
-      && !mappingProfile.getRecordTypes().contains(RecordType.INSTANCE)) {
-      SrsLoadResult srsLoadResult = loadSrsMarcRecordsInPartitions(identifiers, exportPayload.getJobExecutionId(), params);
-      LOGGER.info("Records that are not present in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs());
-      List<String> marcToExport = srsRecordService.transformSrsRecords(mappingProfile, srsLoadResult.getUnderlyingMarcRecords(),
-        exportPayload.getJobExecutionId(), params);
-      exportService.exportSrsRecord(marcToExport, fileExportDefinition);
-      LOGGER.info("Number of instances not found in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs().size());
-      mappingProfileService.getDefault(params.getTenantId())
-        .onSuccess(defaultMappingProfile -> {
-          generateRecordsOnTheFly(exportPayload, identifiers, fileExportDefinition, defaultMappingProfile, params, srsLoadResult);
-          blockingPromise.complete();
-        })
-        .onFailure(ar -> {
-          LOGGER.error("Failed to fetch default mapping profile");
-          errorLogService.saveGeneralError(ErrorCode.DEFAULT_MAPPING_PROFILE_NOT_FOUND.getDescription(), exportPayload.getJobExecutionId(), params.getTenantId());
-          throw new ServiceException(HttpStatus.HTTP_INTERNAL_SERVER_ERROR, ErrorCode.DEFAULT_MAPPING_PROFILE_NOT_FOUND);
-        });
+    if (Objects.isNull(mappingProfile)) {
+      errorLogService.saveGeneralError(ErrorCode.MAPPING_PROFILE_MISSING.getCode(), exportPayload.getJobExecutionId(), params.getTenantId());
+      throw new ServiceException(HttpStatus.HTTP_INTERNAL_SERVER_ERROR, ErrorCode.MAPPING_PROFILE_MISSING);
+    }
+
+    if (mappingProfile.getRecordTypes().contains(RecordType.SRS)) {
+      if (!mappingProfile.getRecordTypes().contains(RecordType.INSTANCE)) {
+        SrsLoadResult srsLoadResult = loadSrsMarcRecordsInPartitions(identifiers, exportPayload.getJobExecutionId(), params);
+        LOGGER.info("Records that are not present in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs());
+        List<String> marcToExport = srsRecordService.transformSrsRecords(mappingProfile, srsLoadResult.getUnderlyingMarcRecords(),
+          exportPayload.getJobExecutionId(), params);
+        exportService.exportSrsRecord(marcToExport, fileExportDefinition);
+        LOGGER.info("Number of instances not found in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs().size());
+        mappingProfileService.getDefault(params.getTenantId())
```

Comment:
This logic is not needed if all the records are found in SRS isn't it? I think a check needs to be added

Reply:
done